syntax = "proto3";

package proto;

message SVCount {
    map<uint64, uint32> counts = 1;
}

message LabelIndex {
    map<uint64, SVCount> blocks = 1;  // key is encoded block coord ZYX (packed little-endian 21-bit numbers where MSB is sign flag)
    uint64 label = 2;
    uint64 last_mut_id = 3;
    string last_mod_time = 4;  // string is time in RFC 3339 format
    string last_mod_user = 5;
    string last_mod_app = 6;
}